File: src/backend/music-server/music-api/src/main/java/com/lalala/music/config/RestClientConfig.java
package com.lalala.music.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.client.RestClient;

@Configuration
public class RestClientConfig {
    @Value("${uploader.url}")
    private String uploaderURL;

    @Bean(name = {"uploaderClient"})
    public RestClient restClient() {
        return RestClient.builder().baseUrl(uploaderURL).build();
    }
}

File: src/backend/streaming-server/src/main/kotlin/com/lalala/streaming/config/client/RestClientConfig.kt
package com.lalala.streaming.config.client

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.stereotype.Component
import org.springframework.web.client.RestClient

@Component
class RestClientConfig(
    @Value("\${client.endpoint.music:http://localhost:24000}") val musicUrl: String,
    @Value("\${client.endpoint.storage:http://localhost:30000}") val storageUrl: String
) {
    @Bean(name = ["musicClient"])
    fun musicRestClient() = RestClient.builder()
        .baseUrl(musicUrl)
        .build()

    @Bean(name = ["storageClient"])
    fun storageRestClient() = RestClient.builder()
        .baseUrl(storageUrl)
        .build()
}
